TestControllers/TestController.psm1

```diff
@@ -726,6 +726,10 @@ Class TestController {
 				$executionCount += 1
 				Write-LogInfo "($executionCount/$($this.TotalCaseNum)) testing started: $($currentTestCase.testName)"
 				Write-LogInfo "SetupConfig: { $(ConvertFrom-SetupConfig -SetupConfig $currentTestCase.SetupConfig) }"
+				# Retrieve VM info, when the VM doesn't have IP, we create new environemnt.
+				if ($vmData -and $this.TestPlatform -imatch "HyperV") {
```

Comment:
It's ok to retrieve IP again for all platforms. It won't be a big cost, and can make all logic consistent.